glossary/milestone.md

```diff
@@ -0,0 +1 @@
+Milestones are specific points in the development process of a [product](product.md) that have a [definition of done](../define-done/readme.md).
```

Comment:
Haha, I just copy pasta'd this from the project. I'll try to make it a bit clearer.